src/org/labkey/test/components/ui/search/SaveSampleFinderViewModal.java

```diff
@@ -0,0 +1,87 @@
+package org.labkey.test.components.ui.search;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class SaveSampleFinderViewModal extends ModalDialog
+{
+    public SaveSampleFinderViewModal(WebDriver driver)
+    {
+        this("Save Custom Search", driver);
+    }
+
+    protected SaveSampleFinderViewModal(String title, WebDriver driver)
+    {
+        super(new ModalDialog.ModalDialogFinder(driver).withTitle(title));
+    }
+
+    public String getName()
+    {
+        return Locator.tag("input").findElement(getComponentElement()).getAttribute("value");
+    }
+
+    public SaveSampleFinderViewModal setName(String name)
+    {
+        WebElement input = elementCache().nameInput;
+        input.clear();
+        if (name != null)
+            input.sendKeys(name);
+        return this;
+    }
+
+    public void clickSave()
+    {
+        clickSave(false);
+    }
+
+    public String clickSave(boolean hasError)
+    {
+        elementCache().saveBtn.click();
+        if (!hasError)
+        {
+            waitForClose();
+            return null;
+        }
+        return getErrorMsg();
+    }
```

Comment:
I believe the pattern we are trying to make the standard in test automation is to have a method named `clickSaveExpectingError()` rather than have an overload that takes a parameter. Admittedly we are not very consistent, but I believe that is the goal. 

These two methods could become:
```
    public void clickSave()
    {
        dismiss("Save");
    }

    public String clickSaveExpectingError()
    {
        elementCache().saveBtn.click();
        return getErrorMsg();
    }

```